lightning/src/ln/channelmanager.rs

```diff
@@ -618,6 +650,8 @@ pub struct ChannelManager<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref,
 	#[cfg(not(any(test, feature = "_test_utils")))]
 	channel_state: Mutex<ChannelHolder<Signer>>,
 
+	fake_last_hops: Mutex<HashMap<u64, ExpandedPmtKey>>,
```

Comment:
Hmmm, not really a fan of storing these if we can avoid it. How about when we generate an invoice we create a random SCID, then use that to derive a random node key? Of course I think some users will probably want a static key, but we can just make that an option? Like, if you want a static node key then we just use a static SCID too and derive something from the key data?